crates/red_knot_python_semantic/resources/mdtest/type_properties/is_subtype_of.md

```diff
@@ -0,0 +1,232 @@
+static_assert(is_subtype_of(Literal[b"foo"], object))
+```
+
+## Tuple types
+
+```py
+from typing_extensions import Literal
+from knot_extensions import is_subtype_of, static_assert
+
+static_assert(is_subtype_of(tuple[()], tuple[()]))
+static_assert(is_subtype_of(tuple[Literal[42]], tuple[int]))
+static_assert(is_subtype_of(tuple[Literal[42], Literal["foo"]], tuple[int, str]))
+static_assert(is_subtype_of(tuple[int, Literal["foo"]], tuple[int, str]))
+static_assert(is_subtype_of(tuple[Literal[42], str], tuple[int, str]))
+
+static_assert(not is_subtype_of(tuple[()], tuple[Literal[1]]))
+static_assert(not is_subtype_of(tuple[Literal[42]], tuple[str]))
+
+static_assert(not is_subtype_of(tuple[tuple[int, ...]], tuple[Literal[2]]))
+static_assert(not is_subtype_of(tuple[Literal[2]], tuple[tuple[int, ...]]))
+```
+
+## Union types
+
+```py
+from typing_extensions import Literal
+from knot_extensions import is_subtype_of, static_assert
+
+static_assert(is_subtype_of(Literal[1], int | str))
+static_assert(is_subtype_of(str | int, object))
+static_assert(is_subtype_of(Literal[1, 2], Literal[1, 2, 3]))
+```
+
+## Intersection types
+
+```py
+from typing_extensions import Literal, LiteralString
+from knot_extensions import Intersection, Not, is_subtype_of, static_assert
+
+static_assert(is_subtype_of(Intersection[int, Not[Literal[2]]], int))
+static_assert(is_subtype_of(Intersection[int, Not[Literal[2]]], Intersection[Not[Literal[2]]]))
+static_assert(is_subtype_of(Intersection[Not[int]], Intersection[Not[Literal[2]]]))
+static_assert(is_subtype_of(Literal[1], Intersection[int, Not[Literal[2]]]))
+static_assert(is_subtype_of(Intersection[str, Not[Literal["foo"]]], Intersection[Not[Literal[2]]]))
+static_assert(is_subtype_of(Intersection[Not[LiteralString]], object))
+
+static_assert(is_subtype_of(type[str], Intersection[Not[None]]))
+static_assert(is_subtype_of(Intersection[Not[LiteralString]], object))
+
+static_assert(not is_subtype_of(Intersection[int, Not[Literal[2]]], Intersection[int, Not[Literal[3]]]))
+static_assert(not is_subtype_of(Intersection[Not[2]], Intersection[Not[3]]))
+static_assert(not is_subtype_of(Intersection[Not[2]], Intersection[Not[int]]))
+
+static_assert(not is_subtype_of(int, Intersection[Not[3]]))
+static_assert(not is_subtype_of(Literal[1], Intersection[int, Not[1]]))
+```
+
+## Class literal types
+
+```py
+from abc import ABC, ABCMeta
+from types import ModuleType
+from typing import _SpecialForm
+from typing_extensions import Literal, LiteralString
+import typing
+from knot_extensions import TypeOf, is_subtype_of, static_assert
+
+static_assert(is_subtype_of(TypeOf[bool], type[int]))
+static_assert(is_subtype_of(TypeOf[int], TypeOf[int]))
+static_assert(is_subtype_of(TypeOf[int], object))
+
+static_assert(is_subtype_of(TypeOf[Literal], _SpecialForm))
+static_assert(is_subtype_of(TypeOf[Literal], object))
+
+static_assert(is_subtype_of(TypeOf[ABC], ABCMeta))
+static_assert(is_subtype_of(ABCMeta, type[object]))
```

Comment:
This comment by @AlexWaygood probably also applies here: https://github.com/astral-sh/ruff/pull/15353#discussion_r1907236295. We could introduce something like
```
class Meta(type): ...
class HasCustomMetaClass(metaclass=Meta): ...
```
and then use those instead of `ABCMeta`, `ABC`.